provider/lxd/server.go

```diff
@@ -253,9 +273,48 @@ func (s *serverFactory) bootstrapLocalServer(svr Server) (Server, string, error)
 			hostAddress, connInfoAddresses,
 		)
 	}
+
+	// If the server is not a simple simple stream server, don't check the
+	// API version, but do report for other scenarios
+	if connInfoProtocol != string(lxd.SimpleStreamsProtocol) {
+		if err := s.validateServer(svr, profile); err != nil {
+			return nil, "", errors.Trace(err)
+		}
+	}
+
 	return svr, hostAddress, nil
 }
 
+func (s *serverFactory) bootstrapRemoteServer(svr Server) error {
+	// We need to get a default profile, so that the local bridge name
+	// can be discovered correctly to then get the host address.
+	defaultProfile, profileETag, err := svr.GetProfile("default")
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	if err := s.validateServer(svr, apiProfile{
+		Profile: defaultProfile,
+		ETag:    profileETag,
+	}); err != nil {
+		return errors.Trace(err)
+	}
+
+	return nil
+}
+
+func (s *serverFactory) validateServer(svr Server, profile apiProfile) error {
```

Comment:
When we are bootstrapping locally, we do want to call `EnsureDefaultNetworking` and `EnsureDefaultStorage`. This is probably OK but _variations_ of these methods _might_ need to be used on remote servers.

We definitely want to check that there is a usable network device and storage set-up, not sure that we want to go modifying the profiles if the servers are set up in a certain fashion.

This is fine for now though, we can cross that bridge when it sees some action in-theatre.